tests/darcy_velocity.prm

```diff
@@ -0,0 +1,148 @@
+#########################################################
+# This is a test for the new advection method that advects a
+# compositional field with the fluid velocity prescribed by
+# Darcy's Law.
+
+# The initial porosity is uniform everywhere, and in the
+# center of the model is a blob of 'fluid' and 'solid'.
+# The solid has a prescribed x velocity, and so the fluid
+# blob should be advected horizontally with the solid and
+# vertically as it rises up due to its buoyancy.
+
+############### Global parameters
+
+set Dimension                              = 2
+set Start time                             = 0
+set End time                               = 5000
+set Use years in output instead of seconds = true
+set Adiabatic surface temperature          = 1600
+set Nonlinear solver scheme                = iterated Advection and Stokes
+set Max nonlinear iterations               = 5
+set CFL number                             = 0.2
+set Output directory                       = darcy_velocity
+set Pressure normalization                 = surface
+set Surface pressure                       = 0
+set Maximum time step                      = 1000
+
+############### Parameters describing the model
+# Let us here choose again a box domain
+# where we fix the temperature at the bottom and top,
+# allow free slip along the boundaries
+# and include melt migration.
+
+subsection Geometry model
+  set Model name = box
+
+  subsection Box
+    set X extent = 200000
+    set Y extent = 100000
+    set X repetitions = 2
+  end
+end
+
+subsection Boundary temperature model
+  set Fixed temperature boundary indicators   = 2, 3
+end
+
+subsection Boundary velocity model
+  set Prescribed velocity boundary indicators = left x:function, right x:function
+  set Tangential velocity boundary indicators = top, bottom
+  subsection Function
+    set Coordinate system = cartesian
+    set Variable names = x,y,t
+    set Function expression = 1;0
+  end
+end
+
+############### Compositional fields
+# We want to use three compositional fields, the porosity and
+# and two additional fields. The solid field will be advected with
+# the solid speed, and the fluid field will be advected as the sum of
+# the solid velocity and the vertical buoyancy the fluid has. The
+# porosity field is uniform to allow for a uniform fluid velocity.
+subsection Compositional fields
+  set Number of fields = 3
+  set Names of fields = porosity, solid, fluid
```

Comment:
Do we technically need to have the solid field, or can we just use the "background" to represent the solid phase? This might be a melt material model feature I missed.

Reply:
The solid field is not needed no, I can remove it. I just had it initially to make sure that the fluid was actually moving with both the solid velocity and the buoyancy